drivers/sdhc/ifx_cat1_sdio.c

```diff
@@ -0,0 +1,341 @@
+{
+	return 1;
+}
+
+static int ifx_cat1_sdio_get_host_props(const struct device *dev, struct sdhc_host_props *props)
+{
+	memset(props, 0, sizeof(*props));
+	props->f_max = IFX_CAT1_SDIO_F_MAX;
+	props->f_min = IFX_CAT1_SDIO_F_MIN;
+	props->host_caps.bus_4_bit_support = true;
+	props->host_caps.high_spd_support = true;
+	props->host_caps.sdr50_support = true;
+	props->host_caps.sdio_async_interrupt_support = true;
+	props->host_caps.vol_330_support = true;
+
+	return 0;
+}
+
+static int ifx_cat1_sdio_enable_interrupt(const struct device *dev, sdhc_interrupt_cb_t callback,
+					  int sources, void *user_data)
+{
+	struct ifx_cat1_sdio_data *data = dev->data;
+	const struct ifx_cat1_sdio_config *cfg = dev->config;
+
+	if (callback == NULL) {
+		return -EINVAL;
+	}
+
+	/* Record SDIO callback parameters */
+	data->sdio_cb = callback;
+	data->sdio_cb_user_data = user_data;
+
+	/* Enable CARD INTERRUPT event */
+	if (sources == SDHC_INT_SDIO) {
+		cyhal_sdio_enable_event(&data->sdio_obj, CYHAL_SDIO_CARD_INTERRUPT,
+					cfg->irq_priority, true);
+	} else {
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int ifx_cat1_sdio_disable_interrupt(const struct device *dev, int sources)
+{
+	struct ifx_cat1_sdio_data *data = dev->data;
+	const struct ifx_cat1_sdio_config *cfg = dev->config;
+
+	data->sdio_cb = NULL;
+	data->sdio_cb_user_data = NULL;
+
+	/* Disable CARD INTERRUPT event */
+	if (sources == SDHC_INT_SDIO) {
+		cyhal_sdio_enable_event(&data->sdio_obj, CYHAL_SDIO_CARD_INTERRUPT,
+					cfg->irq_priority, false);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static void ifx_cat1_sdio_event_callback(void *callback_arg, cyhal_sdio_event_t event)
+{
+	const struct device *dev = callback_arg;
+	struct ifx_cat1_sdio_data *data = dev->data;
+
+	if ((event == CYHAL_SDIO_CARD_INTERRUPT) && (data->sdio_cb != NULL)) {
+		data->sdio_cb(dev, SDHC_INT_SDIO, data->sdio_cb_user_data);
+	}
+}
+
+static int ifx_cat1_sdio_init(const struct device *dev)
+{
+	cy_rslt_t ret;
+	struct ifx_cat1_sdio_data *data = dev->data;
+	const struct ifx_cat1_sdio_config *config = dev->config;
+
+	/* Configure dt provided device signals when available */
+	ret = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (ret) {
+		return ret;
+	}
+
+	/* Dedicate SCB HW resource */
+	data->hw_resource.type = CYHAL_RSC_SDIODEV;
+	data->hw_resource.block_num = _get_hw_block_num(config->reg_addr);
+
+	/* Initialize the SDIO peripheral */
+	data->cyhal_sdio_config.resource = &data->hw_resource;
+	data->cyhal_sdio_config.host_config = &host_config,
+	data->cyhal_sdio_config.card_config = &sd_host_sd_card_config,
+
+	ret = cyhal_sdio_init_cfg(&data->sdio_obj, &data->cyhal_sdio_config);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("cyhal_sdio_init_cfg failed ret = %d \r\n", ret);
+		return ret;
+	}
+
+	/* Register callback for SDIO events */
+	cyhal_sdio_register_callback(&data->sdio_obj, ifx_cat1_sdio_event_callback, (void *)dev);
+
+	k_mutex_init(&data->access_mutex);
```

Comment:
k_mutex_init returns a value, need to check for success/failure.

Reply:
updated